File: 100Doors/100doors.js
function getFinalOpenedDoors(numDoors) {
  const doors = [];
  const open_doors = [];
  for (let i = 0; i < numDoors; i++){
    doors.push(false);
  }
  let tracker = 1;
  while (tracker <= numDoors){
    for (let i = tracker - 1; i < numDoors; i += tracker){
      doors[i] = !doors[i];
    }
    tracker += 1;
  }
  
  for (let i = 0; i < numDoors; i++){
    if (doors[i]){
      open_doors.push(i + 1);
    }
  }
  return open_doors;
}

if(require.main === module){
  const numDoors = parseInt(process.argv[2], 10);
  if (isNaN(numDoors)){
    console.error();
    process.exit(1);
  }
  const result = getFinalOpenedDoors(numDoors);
  console.log(result);
}

File: 100Doors/100doors.go
package main

func getFinalOpenedDoors(numDoors int) []int {
	var doors []bool = make([]bool, numDoors)
	var open_doors []int

	for i := 0; i < numDoors; i++ {
		doors[i] = false
	}

	var tracker int = 1
	for tracker <= numDoors {
		for i := tracker - 1; i < numDoors; i += tracker {
			doors[i] = !doors[i]
		}
		tracker += 1
	}

	for i := 0; i < numDoors; i++ {
		if doors[i] {
			open_doors = append(open_doors, i+1)
		}
	}
	return open_doors

}
